client/src/main/java/com/alibaba/nacos/client/security/SecurityProxy.java

```diff
@@ -112,11 +118,24 @@ public boolean login(String server) {
             Map<String, String> params = new HashMap<String, String>(2);
             params.put("username", username);
             String body = "password=" + password;
-            String url = "http://" + server + contextPath + LOGIN_URL;
-
-            if (server.contains(Constants.HTTP_PREFIX)) {
-                url = server + contextPath + LOGIN_URL;
+            String url=server;
+            if (!server.startsWith(Constants.HTTP_PREFIX)) {
+                if (HttpClient.ENABLE_HTTPS) {
+                    url = "https://" + server + contextPath + LOGIN_URL;
+                } else {
+                    url = "http://" + server + contextPath + LOGIN_URL;
+                }
+            }else{
+                if (HttpClient.ENABLE_HTTPS && !server.startsWith(Constants.HTTPS_PREFIX)) {
+                    url =Constants.HTTPS_PREFIX +StringUtils.removeStart(server,Constants.HTTP_PREFIX)+ contextPath + LOGIN_URL;
+                } else {
+                    url = server + contextPath + LOGIN_URL;
+                }
```

Comment:
Here, there are some String character processing, you may use some more optimized methodsm, such as StringBuilder.appened()?